bug: java.lang.NoSuchMethodException with native compilation and ClientRequestFilter for Ollama

Hi,
I followed the [documentation](https://docs.quarkiverse.io/quarkus-langchain4j/dev/ollama.html) to add a request filter to handle authentication for the Ollama models.

Everything  work fine when I run the program in the standard JAva mode but I'v got an `java.lang.NoSuchMethodException` when I use the native application built with `quarkus build --native`.

I took the same code as mentionned in the documentation:
```java
@Provider
public class OllamaClientAuthHeaderFilter implements ClientRequestFilter {

    @Override
    public void filter(ClientRequestContext requestContext) {
        requestContext.getHeaders().add("Authorization", "Bearer " + System.getenv("OVH_OLLAMA_API_KEY"));
    }
}
```

The stacktrace:
```bash
jakarta.enterprise.inject.CreationException: Error creating synthetic bean [FwyQP9Of9oZwwZhlQz1A4k1Ak7I]: java.lang.RuntimeException: java.lang.NoSuchMethodException: fr.wilda.picocli.sdk.ai.OllamaClientAuthHeaderFilter.<init>()
        at dev.langchain4j.model.chat.StreamingChatLanguageModel_FwyQP9Of9oZwwZhlQz1A4k1Ak7I_Synthetic_Bean.doCreate(Unknown Source)
        at dev.langchain4j.model.chat.StreamingChatLanguageModel_FwyQP9Of9oZwwZhlQz1A4k1Ak7I_Synthetic_Bean.create(Unknown Source)
        at dev.langchain4j.model.chat.StreamingChatLanguageModel_FwyQP9Of9oZwwZhlQz1A4k1Ak7I_Synthetic_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:38)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:35)
        at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.c10(Unknown Source)
        at io.quarkus.arc.generator.Default_jakarta_enterprise_context_ApplicationScoped_ContextInstances.computeIfAbsent(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:35)
        at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:21)
        at dev.langchain4j.model.chat.StreamingChatLanguageModel_FwyQP9Of9oZwwZhlQz1A4k1Ak7I_Synthetic_ClientProxy.arc$delegate(Unknown Source)
        at dev.langchain4j.model.chat.StreamingChatLanguageModel_FwyQP9Of9oZwwZhlQz1A4k1Ak7I_Synthetic_ClientProxy.generate(Unknown Source)
        at dev.langchain4j.service.AiServiceTokenStream$AiServiceOnStart.start(AiServiceTokenStream.java:90)
        at io.quarkiverse.langchain4j.runtime.aiservice.AiServiceMethodImplementationSupport$2.accept(AiServiceMethodImplementationSupport.java:172)
        at io.quarkiverse.langchain4j.runtime.aiservice.AiServiceMethodImplementationSupport$2.accept(AiServiceMethodImplementationSupport.java:160)
        at io.smallrye.context.impl.wrappers.SlowContextualConsumer.accept(SlowContextualConsumer.java:21)
        at io.smallrye.mutiny.operators.multi.builders.EmitterBasedMulti.subscribe(EmitterBasedMulti.java:67)
        at io.smallrye.mutiny.operators.AbstractMulti.subscribe(AbstractMulti.java:60)
        at io.smallrye.mutiny.helpers.BlockingIterable.lambda$stream$1(BlockingIterable.java:59)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:604)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:11)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:11)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base@21.0.2/java.lang.Thread.runWith(Thread.java:1596)
        at java.base@21.0.2/java.lang.Thread.run(Thread.java:1583)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:833)
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:211)
Caused by: java.lang.RuntimeException: java.lang.NoSuchMethodException: fr.wilda.picocli.sdk.ai.OllamaClientAuthHeaderFilter.<init>()
        at org.jboss.resteasy.reactive.common.jaxrs.ConfigurationImpl.register(ConfigurationImpl.java:211)
        at org.jboss.resteasy.reactive.client.impl.ClientBuilderImpl.register(ClientBuilderImpl.java:351)
        at io.quarkus.rest.client.reactive.runtime.RestClientBuilderImpl.register(RestClientBuilderImpl.java:242)
        at io.quarkus.rest.client.reactive.runtime.RestClientBuilderImpl.build(RestClientBuilderImpl.java:352)
        at io.quarkus.rest.client.reactive.runtime.QuarkusRestClientBuilderImpl.build(QuarkusRestClientBuilderImpl.java:260)
        at io.quarkiverse.langchain4j.ollama.OllamaClient.<init>(OllamaClient.java:29)
        at io.quarkiverse.langchain4j.ollama.OllamaStreamingChatLanguageModel.<init>(OllamaStreamingChatLanguageModel.java:28)
        at io.quarkiverse.langchain4j.ollama.OllamaStreamingChatLanguageModel$Builder.build(OllamaStreamingChatLanguageModel.java:147)
        at io.quarkiverse.langchain4j.ollama.runtime.OllamaRecorder$5.get(OllamaRecorder.java:151)
        at io.quarkiverse.langchain4j.ollama.runtime.OllamaRecorder$5.get(OllamaRecorder.java:148)
        at io.quarkus.arc.runtime.ArcRecorder$4.apply(ArcRecorder.java:129)
        at io.quarkus.arc.runtime.ArcRecorder$4.apply(ArcRecorder.java:126)
        at dev.langchain4j.model.chat.StreamingChatLanguageModel_FwyQP9Of9oZwwZhlQz1A4k1Ak7I_Synthetic_Bean.createSynthetic(Unknown Source)
        ... 30 more
Caused by: java.lang.NoSuchMethodException: fr.wilda.picocli.sdk.ai.OllamaClientAuthHeaderFilter.<init>()
        at java.base@21.0.2/java.lang.Class.checkMethod(DynamicHub.java:1075)
        at java.base@21.0.2/java.lang.Class.getConstructor0(DynamicHub.java:1238)
        at java.base@21.0.2/java.lang.Class.getDeclaredConstructor(DynamicHub.java:2930)
        at org.jboss.resteasy.reactive.common.jaxrs.ConfigurationImpl.register(ConfigurationImpl.java:209)
        ... 42 more
```

Do you any idea what I'm doing wrong?

--- a/docs/modules/ROOT/pages/ollama.adoc
+++ b/docs/modules/ROOT/pages/ollama.adoc
@@ -74,6 +74,9 @@ public class OllamaClientAuthHeaderFilter implements ClientRequestFilter {
 
 This class is a CDI bean which gives the ability to utilize other CDI beans if necessary.
 
+NOTE: If you want to use it with native compilation you should add the `@RegisterForReflection` annotation. See https://quarkus.io/guides/writing-native-applications-tips#registering-for-reflection[Quarkus documentation] for more information.  
+This should be not necessary with future releases of Quarkus.
+
 == Document Retriever and Embedding
 
 Ollama also provides embedding models.

